Jenkinsfiles/build.groovy

```diff
@@ -0,0 +1,258 @@
+                currentBuild.result = 'ABORTED';
+                currentBuild.description = "Branch: ${CHANGE_BRANCH} exists on nuxeo/nuxeo repository, aborting build."
+                error(currentBuild.description)
+              }
+            }
+          }
+        }
+      }
+    }
+
+    stage('Update version') {
+      steps {
+        container('maven') {
+          updateVersion()
+        }
+      }
+    }
+
+    stage('Compile') {
+      steps {
+        setGitHubBuildStatus('jsfui/compile', 'Compile', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Compile
+          ----------------------------------------"""
+          echo "MAVEN_OPTS=$MAVEN_OPTS"
+          sh 'mvn -V -B -nsu -N install'
+          sh 'mvn -B -nsu -T0.8C -DskipTests -f code/pom.xml install'
+        }
+      }
+
+      post {
+        success {
+          setGitHubBuildStatus('jsfui/compile', 'Compile', 'SUCCESS')
+        }
+        failure {
+          setGitHubBuildStatus('jsfui/compile', 'Compile', 'FAILURE')
+        }
+      }
+    }
+
+    stage('Run "dev" unit tests') {
+      steps {
+        setGitHubBuildStatus('jsfui/utests/dev', 'Unit tests - dev environment', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Run "dev" unit tests
+          ----------------------------------------"""
+          echo "MAVEN_OPTS=$MAVEN_OPTS"
+          sh "mvn -B -nsu -f code/pom.xml test"
+        }
+      }
+
+      post {
+        always {
+          junit testResults: '**/target/surefire-reports/*.xml'
+        }
+        success {
+          setGitHubBuildStatus('jsfui/utests/dev', 'Unit tests - dev environment', 'SUCCESS')
+        }
+        failure {
+          setGitHubBuildStatus('jsfui/utests/dev', 'Unit tests - dev environment', 'FAILURE')
+        }
+      }
+    }
+
+    stage('Package') {
+      steps {
+        setGitHubBuildStatus('jsfui/package', 'Package', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Package
+          ----------------------------------------"""
+          echo "MAVEN_OPTS=$MAVEN_OPTS"
+          sh 'mvn -B -nsu -T0.8C -DskipTests -f packages/pom.xml install'
+        }
+      }
+
+      post {
+        success {
+          setGitHubBuildStatus('jsfui/package', 'Package', 'SUCCESS')
+        }
+        failure {
+          setGitHubBuildStatus('jsfui/package', 'Package', 'FAILURE')
+        }
+      }
+    }
+
+    stage('Run "dev" functional tests') {
+      steps {
+        setGitHubBuildStatus('jsfui/ftests/dev', 'Functional tests - dev environment', 'PENDING')
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Run "dev" functional tests
+          ----------------------------------------"""
+          script {
+            try {
```

Comment:
why not use post/success/failure?

Reply:
There is no way to set the build as blue if we use "post/success/failure" :(
This `try ... catch` will be removed when functional tests random failures will be handled.